packages/state-transition/src/cache/stateCache.ts

```diff
@@ -126,6 +127,7 @@ export type CachedBeaconStateAltair = CachedBeaconState<BeaconStateAltair>;
 export type CachedBeaconStateBellatrix = CachedBeaconState<BeaconStateBellatrix>;
 export type CachedBeaconStateCapella = CachedBeaconState<BeaconStateCapella>;
 export type CachedBeaconStateDeneb = CachedBeaconState<BeaconStateDeneb>;
+export type CachedBeaconStateEIP6110 = CachedBeaconState<BeaconStateEIP6110>;
```

Comment:
rename `createCachedBeaconState` to `createFinalizedCachedBeaconState` and add comment for it saying this is supposed for finalized state. Could add a TODO and do it in a separate PR.